src/app/component/user/components/habit/all-habits/all-habits.component.ts

```diff
@@ -8,30 +11,88 @@ import { Component, OnInit, OnDestroy } from '@angular/core';
   styleUrls: ['./all-habits.component.scss']
 })
 export class AllHabitsComponent implements OnInit, OnDestroy {
-
-  public habitsMockList: object[];
+  private masterSubscription: Subscription = new Subscription();
+  private habitsList: ServerHabitItemPageModel[] = [];
+  public filteredHabitsList: ServerHabitItemPageModel[] = [];
+  public totalHabits: number;
+  public totalHabitsCopy = 0;
   public galleryView = true;
-  private subscription: Subscription;
+  private lang: string;
   public windowSize: number;
+  private batchSize = 6;
+  private currentPage = 0;
+  private totalPages: number;
+  public isFetching = true;
+  public elementsLeft = true;
+  public tagList: string[] = [];
 
-  constructor(private allHabitsService: AllHabitsService) { }
+  constructor( private allHabitsService: AllHabitsService,
+               private localStorageService: LocalStorageService,
+               private translate: TranslateService ) { }
 
   ngOnInit() {
-    this.getAllHabits();
     this.onResize();
+
+    const langChangeSub = this.localStorageService.languageBehaviourSubject.subscribe(lang => {
+      this.translate.setDefaultLang(lang);
+      this.lang = lang;
+      this.resetState();
+      this.allHabitsService.resetSubject();
+      this.getAllHabits(0, this.batchSize, lang);
+    });
+
+    const habitServeceSub = this.allHabitsService.allHabits.subscribe(data => {
+      this.isFetching = false;
```

Comment:
I haven't found "langChangeSub.unsubscribe" and "habitServeceSub.unsubscribe"

Reply:
this is because i have 1 "parent" subscriber -> masterSubscription
and I add all the subscribers to it
this.masterSubscription.add (langChangeSub);
this.masterSubscription.add (habitServeceSub)

and I can unsubscribe only from 1 "parent" subscriber)
  this.masterSubscription.unsubscribe ();